app/src/forms/submission/service.js

```diff
@@ -0,0 +1,100 @@
+const { Form, FormVersion, FormSubmission, FormSubmissionUserPermissions, SubmissionMetadata, UserFormAccess } = require('../common/models');
+
+const Permissions = require('../common/constants').Permissions;
+
+const Problem = require('api-problem');
+const {transaction} = require('objection');
+
+const service = {
+
+  _fetchSubmissionData: async(formSubmissionId) => {
+    const meta = await SubmissionMetadata.query()
+      .where('submissionId', formSubmissionId)
+      .first()
+      .throwIfNotFound();
+
+    const submission = await FormSubmission.query()
+      .findById(meta.submissionId)
+      .throwIfNotFound();
+    const version = await FormVersion.query()
+      .findById(meta.formVersionId)
+      .throwIfNotFound();
+    const form = await Form.query()
+      .findById(meta.formId)
+      .allowGraph('identityProviders')
+      .withGraphFetched('identityProviders(orderDefault)')
+      .throwIfNotFound();
```

Comment:
Looking at this, may be possible to have these three queries done in a `Promise.all(...)` wrapper so that we can parallelize them. Realistically, these three queries "should" be pretty fast already, but there isn't anything preventing them from being completely asynchronous until the return statement (or if it throws)

Reply:
putting in promise.all(), will create another ticket to clean up the open api spec, no need holding this pr up.